### DISCUSSION TITLE
Playing videos over the internet is extremely slow / constant buffering

### DISCUSSION THREAD

**Discussion by LeehamElectronics** [photoprism/photoprism#3286] [Q&A]
Anytime I share a video album with friends they always complain that the videos are unplayable as they are constantly buffering. This seems to only happen with larger videos above 5 minutes in length. 
I tested this myself from outside my network over the internet and found it to be true.

Of course playing videos of any size from within my LAN works perfectly fine, they play instantly. 

I tried bypassing Cloudflare and my local nginx reverse proxy in case the issue lied somewhere there, it did not. I tried directly viewing a video through the internet to PhotoPrism via it's exposed ports (HTTP, I know, it's not safe) and the issue prevailed.

Therefore, I'm assuming the issue must be Photoprism. I enabled debug and couldn't see any errors in logs or anything related to ffpmeg. The videos I tried playing are all AVC files, so they should be compatible with web clients. I tried testing all web browsers.

My internet connection should not be an issue, it would bottle neck at about 30 mega bits per second. Plex streams work just fine so it shouldn't be related to my internet.

Is there anything else I can do to diagnose this issue? I really need to be able to share videos with people.
Cheers.

**Comment by lastzero** [comment]
We use standard libraries for streaming. So I would be very surprised if there is an arbitrary 5-minute or file size limit for video streaming. It could still be that the bitrate is too high for your internet connection, since you didn't provide any details on the videos themselves. Cloudflare users have also reported issues with videos in the past, which is presumably due to their special video policy:

- https://developers.cloudflare.com/support/more-dashboard-apps/cloudflare-stream/delivering-videos-with-cloudflare/

**Comment by LeehamElectronics** [comment]
Thanks for the reply, I set my max bit rate to a higher number, and I am not using Cloudflare for testing purposes. How can I tell if the video stream is being actually streamed to the phone? Because at the moment it seems as though the client is trying to download the video before playing it, at least it seems that way. 

Is there any other way I can diagnose this issue? My homelab upload speed is 40 megabits per second, so that should be able to push a video stream to one client, especially since Plex runs just fine.

**Reply by lastzero** [reply]
https://support.plex.tv/articles/115007570148-automatically-adjust-quality-when-streaming/

As PhotoPrism primarily is a photo application, it does not support adaptive bitrate transcoding at this time. Note that 50 MBit/s is a typical video bitrate, so a 40 MBit connection would lead to what you describe. If anything you must therefore reduce the bitrate setting in PhotoPrism, not increase it. Note AVC videos will not be reencoded though. See our docs.

**Reply by lastzero** [reply]
40 MB/s, so about 400 Mbit/s upstream? FTTH?

**Reply by LeehamElectronics** [reply]
> 40 MB/s, so about 400 Mbit/s upstream? FTTH?

sorry about that, I must be really tired. I read my original comment wrong haha. Updated it again. Thanks

**Reply by harshavmb** [reply]
I did have the similar problem streaming 4K Videos but I think the problem is largely with ISPs restricting upload speeds. 

I was hosting on a raspberrypi & Videos buffered almost all the time. I blamed on Pi but one fine day when I did cast on LAN network to TV it worked very fine. This gave me a hint that its the ISP. My upload speed is 50Mbps & could see the same on my Fritzbox but the Video quality was very poor.

I migrated the infra to a hosting provider in Germany (around 200 kms far), Since then no more buffering. I'm playing 4K Videos of 10GB file size pulled over the network storage & served to me over Internet. If it's HDD or SSD, it must be even better. But that can be easily verified over LAN.
